Services/AssessmentQuestion/interfaces/interface.ilAsqQuestion.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/* Copyright (c) 1998-2013 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+ * Interface ilAsqQuestion
+ *
+ * @author    Björn Heyser <info@bjoernheyser.de>
+ * @version    $Id$
+ *
+ * @package    Services/AssessmentQuestion
+ */
+interface ilAsqQuestion
+{
+	/**
+	 * @param int $parentId
+	 * @return void
+	 */
+	public function setParentId($parentId);
+	
+	/**
+	 * @return int
+	 */
+	public function getParentId();
+	
+	/**
+	 * @param int $questionId
+	 * @return void
+	 */
+	public function setId($questionId);
+	
+	/**
+	 * @return int $questionId
+	 */
+	public function getId();
+	
+	/**
+	 * @return string
+	 */
+	public function getTitle();
+	
+	/**
+	 * @return string
+	 */
+	public function getComment();
+	
+	/**
+	 * @return string
+	 */
+	public function getOwner();
+	
+	/**
+	 * @return string $questionType
+	 */
+	public function getQuestionType();
+	
+	/**
+	 * @return string
+	 */
+	public function getQuestionText();
+	
+	/**
+	 * @return string
+	 */
+	public function getPoints();
+	
+	/**
+	 * @return string
+	 */
+	public function getEstimatedWorkingTime();
+	
+	/**
+	 * @return void
+	 */
+	public function load();
+	
+	/**
+	 * @return void
+	 */
+	public function save();
+	
+	/**
+	 * @return void
+	 */
+	public function delete();
+	
+	/**
+	 * @param ilQTIItem $qtiItem
+	 * @return void
+	 */
+	public function fromQtiItem(ilQTIItem $qtiItem);
+	
+	/**
+	 * @return string
+	 */
+	public function toQtiXML();
+	
+	/**
+	 * @return bool
+	 */
+	public function isComplete();
+	
+	/**
+	 * @return ilAsqQuestionSolution
+	 */
+	public function getBestSolution();
+	
+	/**
+	 * @return ILIAS\UI\Component\Legacy\Legacy
+	 */
+	public function getSuggestedSolutionOutput();
+	
+	/**
+	 * @return string
+	 */
+	public function toJSON();
+	
+	/**
+	 * @param string $offlineExportImagePath
+	 */
+	public function setOfflineExportImagePath($offlineExportImagePath = null);
```

Comment:
What is that supposed to do?

Reply:
In the current learning module integration the learning module needs to set an image path to the question instance because there will be no real reimplementation of this presentation type.

In future there should be a any kind of new Presentation inteface, or all this needs to be added to the current one. But I must not resolve the problem of two different presentation implementation during extracting the service, because there are a hell of technical problems that need to be clarified in a desired concept first. Concepts about this were discussed in a group with the SIG EA for years now, but no concrete idea did made it into any concept for now.